[Adjusted to apply to Debian's 2.6.32 by dann frazier <dannf@debian.org>]

commit 9e8342971d44ce86d8567047f5366fc1c06a75ed
Author: Hagen Paul Pfeifer <hagen@jauu.net>
Date:   Wed Oct 7 14:43:04 2009 -0700

    econet: Fix redeclaration of symbol len
    
    Function argument len was redeclarated within the
    function. This patch fix the redeclaration of symbol 'len'.
    
    Signed-off-by: Hagen Paul Pfeifer <hagen@jauu.net>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff -urpN linux-source-2.6.32.orig/net/econet/af_econet.c linux-source-2.6.32/net/econet/af_econet.c
--- linux-source-2.6.32.orig/net/econet/af_econet.c	2010-11-24 16:46:40.911118926 -0700
+++ linux-source-2.6.32/net/econet/af_econet.c	2010-11-24 16:52:20.023304585 -0700
@@ -447,15 +447,15 @@ static int econet_sendmsg(struct kiocb *
 	iov[0].iov_len = size;
 	for (i = 0; i < msg->msg_iovlen; i++) {
 		void __user *base = msg->msg_iov[i].iov_base;
-		size_t len = msg->msg_iov[i].iov_len;
+		size_t iov_len = msg->msg_iov[i].iov_len;
 		/* Check it now since we switch to KERNEL_DS later. */
-		if (!access_ok(VERIFY_READ, base, len)) {
+		if (!access_ok(VERIFY_READ, base, iov_len)) {
 			mutex_unlock(&econet_mutex);
 			return -EFAULT;
 		}
 		iov[i+1].iov_base = base;
-		iov[i+1].iov_len = len;
-		size += len;
+		iov[i+1].iov_len = iov_len;
+		size += iov_len;
 	}
 
 	/* Get a skbuff (no data, just holds our cb information) */
